lxd/migration/migration_volumes.go

```diff
@@ -205,7 +205,10 @@ func MatchTypes(offer *MigrationHeader, fallbackType MigrationFSType, ourTypes [
 				}
 			}
 
-			if offer.Refresh != nil && *offer.Refresh {
+			// Only check for optimized refresh features if refreshing.
+			// It should also be considered that offer.Refresh can be nil for older LXD versions.
+			// In that case, optimized refresh won't be done using zfs or btrfs.
+			if refresh && (offer.Refresh == nil || offer.Refresh != nil && *offer.Refresh) {
```

Comment:
We can probably simplify this check by using offer.GetRefresh() https://github.com/monstermunchkin/lxd/blob/3148444d319736f743215afd5220fabedf281851/lxd/migration/migrate.pb.go#L746